### DISCUSSION TITLE
Is there a way to get the plugin to work in AWS CodeCommit?

### DISCUSSION THREAD

**Discussion by JZWong** [ProjectBorealis/UEGitPlugin#164] [Q&A]
Like the title says, Is there a way to get the plugin to work in AWS CodeCommit? I am trying to hook this plugin up to my AWS code commit, but it does not seem to want to checkout any files.
```
LogSlate: Window 'Revision Control Login' being destroyed
LogSourceControl: Git plugin '3.15'
LogSourceControl: Using 'C:\Program Files\Git\bin\git.exe'
LogSlate: Window 'Revision Control Login' being destroyed
SourceControl: Error: Locking Content/Characters/BP_BaseController.uasset failed: Locking Content/Characters/BP_BaseController.uasset failed: exit status 1
LogSlate: Window 'Message' being destroyed
Message dialog closed, result: Ok, title: Message, text: Git command failed. Please check your connection and try again, or check the output log for more information.
LogSourceControl: Error: Command 'CheckOut' Failed!
LogSlate: Window 'Unable to Check Out From Revision Control!' being destroyed
Message dialog closed, result: Ok, title: Unable to Check Out From Revision Control!, text: The following assets could not be successfully checked out from revision control:
/Game/Characters/BP_BaseController
```

**Comment by mastercoms** [comment]
does it support LFS? if not, disable it in settings

**Reply by JZWong** [reply]
Does Code Commit support LFS? It looks like it does since I was able to commit LFS files, but idk if it can lock files. Where would I disable LFS? Doesn't this plugin require LFS?

**Reply by mastercoms** [reply]
The plugin doesn't require LFS, you can disable it in source control settings, uncheck Git LFS locking

**Reply by JZWong** [reply]
I kind of figured, but I actually do want the locking. So as of right now my assumption is that AWS CodeCommit doesn't support file locking. Would you agree?

**Reply by mastercoms** [reply]
I think so, I can't find any official information about LFS with CodeCommit anywhere in general so I can't know for sure.

**Reply by JZWong** [reply]
Gotcha. Maybe I'll just use it as a submodule for my meshes. Do you know if your plugin Cascades into submodules? I basically want to do the whole checking out and file locking in my main repo, but not have the ability to do any of that in my submodules. Is that possible to do?

**Reply by mastercoms** [reply]
Yeah, I am pretty sure we have submodule support. It isn't a tested path for us so there might be bugs, but I am pretty sure other people in the community have gotten good use out of it for Plugins at least, not sure about the Content folder, but it might work.
